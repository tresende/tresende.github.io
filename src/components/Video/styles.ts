import styled from 'styled-components'

export const Wrapper = styled.video`
  position: fixed;
  left: 0;
  top: 0;
  min-width: 100%;
  min-height: 100%;
`
