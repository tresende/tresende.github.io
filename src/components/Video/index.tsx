import React from 'react'

import * as S from './styles'

const Bio = () => {
  return (
    <S.Wrapper preload="none" autoPlay muted loop>
      <source
        src="https://raw.githubusercontent.com/tresende/tresende.github.io/master/public/videos/bg.mp4"
        type="video/mp4"
      />
    </S.Wrapper>
  )
}
export default Bio
