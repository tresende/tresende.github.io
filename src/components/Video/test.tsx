import { render, screen } from '@testing-library/react'

import Video from '.'

describe('<Video />', () => {
  it('should render background video', () => {
    const { container } = render(<Video />)
    const sourceElement = container.querySelector('source')

    expect(sourceElement?.getAttribute('src')).toBe('/videos/bg.mp4')
    expect(sourceElement?.getAttribute('type')).toBe('video/mp4')
  })

  it('should render image fallaback', () => {
    const { container } = render(<Video />)

    expect(screen.getByRole('img', { name: 'busy avenue' })).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
