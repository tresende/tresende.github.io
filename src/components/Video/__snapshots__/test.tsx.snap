// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Video /> should render image fallaback 1`] = `
<div>
  <video
    autoplay=""
    class="fixed w-auto min-w-full min-h-full max-w-none"
    loop=""
    playsinline=""
    preload="auto"
  >
    <img
      alt="busy avenue"
      src="/img/fake-bg.png"
    />
    <source
      src="/videos/bg.mp4"
      type="video/mp4"
    />
  </video>
</div>
`;
